syndeo-template/src/test/java/com/google/cloud/syndeo/it/KafkaToBigQueryIT.java

```diff
@@ -0,0 +1,347 @@
+            .setJobId(syndeoPipeline.jobId())
+            .setProject(PROJECT)
+            .setRegion(REGION)
+            .build(),
+        () -> {
+          try {
+            DataflowTemplateClient.JobState jobState =
+                templateClient.getJobStatus(PROJECT, REGION, syndeoPipeline.jobId());
+            return jobState.equals(DataflowTemplateClient.JobState.RUNNING);
+          } catch (IOException e) {
+            // If we were unable to check on the job, then we exit the loop.
+            throw new RuntimeException(e);
+          }
+        });
+    // Sleep while the pipeline runs to move all the data.
+    Thread.sleep(3 * ONE_MINUTE_MILLIS);
+
+    // Wait for a while to check BigQuery results
+    {
+      Instant start = Instant.now();
+      while (true) {
+        Long rowsInTable =
+            bigQueryResourceManager.getRowCount(
+                PROJECT,
+                BIGQUERY_DATASET,
+                BIGQUERY_TABLE.substring(BIGQUERY_TABLE.lastIndexOf(".") + 1));
+        if (rowsInTable == ELEMENTS_PER_SECOND * PIPELINE_DURATION_SECONDS) {
+          // Test passes yay!
+          break;
+        } else if (rowsInTable > ELEMENTS_PER_SECOND * PIPELINE_DURATION_SECONDS) {
+          throw new AssertionError(
+              String.format(
+                  "Expected at most %s elements to be inserted to BigQuery, but found %s.",
+                  ELEMENTS_PER_SECOND * PIPELINE_DURATION_SECONDS, rowsInTable));
+        } else if (java.time.Duration.between(start, Instant.now())
+                .compareTo(java.time.Duration.ofMinutes(20))
+            > 0) {
+          // If we spent over 20 minutees waiting for the job, then we must exit
+          throw new AssertionError(
+              String.format(
+                  "Expected %s elements to be inserted to BigQuery, but found %s after over 20 minutes.",
+                  ELEMENTS_PER_SECOND * PIPELINE_DURATION_SECONDS, rowsInTable));
+        } else {
+          // Iterate once more
+          Thread.sleep(ONE_MINUTE_MILLIS);
+          continue;
+        }
+      }
+    }
+
+    templateClient.drainlJob(PROJECT, REGION, syndeoPipeline.jobId());
+    // Wait three minutes while the pipeline drains
+    operator.waitForCondition(
+        DataflowOperator.Config.builder()
+            .setTimeoutAfter(java.time.Duration.ofMinutes(3))
+            .setJobId(syndeoPipeline.jobId())
+            .setProject(PROJECT)
+            .setRegion(REGION)
+            .build(),
+        () -> {
+          try {
+            DataflowTemplateClient.JobState jobState =
+                templateClient.getJobStatus(PROJECT, REGION, syndeoPipeline.jobId());
+            return DataflowTemplateClient.JobState.DONE_STATES.contains(jobState);
+          } catch (IOException e) {
+            // If we were unable to check on the job, then we exit the loop.
+            throw new RuntimeException(e);
+          }
+        });
+  }
+
+  private void createKafkaTopic()
```

Comment:
We should maybe move this to a KafkaResouceManager class? Not sure if you already considered this and plan to refactor this later

Reply:
added a TODO if that's okay? : D